File: JS/523-continuous-subarray-sum.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {boolean}
 */
var checkSubarraySum = function (nums, k) {
  let map = new Map();
  let sum = 0,
    rem;
  map.set(0, -1);
  for (let [i, num] of nums.entries()) {
    sum += num;
    let rem = sum % k;
    let val = map.get(rem);
    if (val != undefined) {
      if (i - val >= 2) return true;
    } else {
      map.set(rem, i);
    }
  }
  return false;
};

File: C++/523-continuous-subarray-sum.cpp
#include "header.h"

class Solution {
public:
    bool checkSubarraySum(vector<int>& nums, int k) {
        unordered_map<int, int> umap;
        int sum = 0;
        umap[0] = 0;
        for (int i = 0; i < nums.size(); i++) {
            sum += nums[i];
            int rem = sum % k;
            if (!umap.count(rem)) umap[rem] = i + 1;
            else if (umap[rem] < i) return true;
        }
        return false;
    }
};